csharp/src/Apache.Arrow/Ipc/ArrowStreamWriter.cs

```diff
@@ -522,13 +522,12 @@ private protected async Task WriteDictionaryAsync(Field field, CancellationToken
             long id = DictionaryMemo.GetId(field);
             IArrowArray dictionary = DictionaryMemo.GetDictionary(id);
 
-            var fieldsDictionary = new Dictionary<string, Field> {
-                { dictionaryField.Name, dictionaryField } };
+            var fieldsList = new List<Field> { dictionaryField };
```

Comment:
```suggestion
            var fields = new Field[] { dictionaryField };
```

Array implements `IReadOnlyList`, so we can just create an array here instead of a List and save on an allocation.

Reply:
Done